dom.bs

```diff
@@ -5956,6 +5956,23 @@ is an object or one of its <a>shadow-including ancestors</a>.
 <p class=note>The <a>retargeting</a> algorithm is used by <a lt=dispatch>event dispatch</a> as well
 as other specifications, such as <cite>Fullscreen</cite>. [[FULLSCREEN]]
 
+<p>To get the <dfn export>flat tree parent</dfn> of a <var>node</var>:</p>
+
+<ol>
+ <li><p>Let <var>parent</var> be <var>node</var>'s <a for=tree>parent</a>.</p></li>
+
+ <li><p>If <var>parent</var> is a <a for=/>shadow root</a>, then return <var>parent</var>'s <a
+ for=DocumentFragment>host</a>.</p></li>
+
+ <li><p>If <var>parent</var> is not a <a for=Element>shadow host</a>, then return
+ <var>node</var>.</p></li>
+
+ <li><p>If <var>node</var>'s <a for=slottable>assigned slot</a> is null, then return null.</p></li>
+
+ <li><p>Return <var>node</var>'s <a for=slottable>assigned slot</a>'s <a
+ for=tree>parent</a>.</p></li>
```

Comment:
@mfreed7 do I need to run anything recursive in the last step here? I briefly tried to make a demo with a slot assigned into another slot, but it didn't seem to work.

Reply:
I guess I at least need to check if the parent is a shadow host, right?